### ISSUE TITLE
CupertinoPicker no longer allows null as backgroundColor

### ISSUE THREAD

**Issue by KammererTob** [flutter/flutter#24392]
Since one of the newest updates (i believe 0.11.3 on the beta branch) CupertinoPicker no longer allows null for backgroundColor although the quick-documentation of the property still indicates that it should be possible.

> The [backgroundColor] defaults to light gray. It can be set to null to
> disable the background painting entirely; this is mildly more efficient
> than using [Colors.transparent].

It might be related to parts of this merged pull-request:
https://github.com/flutter/flutter/pull/24085

## Steps to Reproduce

1. Create a CupertinoPicker while explicitly setting backgroundColor: null

## Logs

```
I/flutter (10259): ══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
I/flutter (10259): The following NoSuchMethodError was thrown building CupertinoPicker(dirty, state:
I/flutter (10259): _CupertinoPickerState#ae960):
I/flutter (10259): The method 'withAlpha' was called on null.
I/flutter (10259): Receiver: null
I/flutter (10259): Tried calling: withAlpha(242)
I/flutter (10259):
I/flutter (10259): When the exception was thrown, this was the stack:
I/flutter (10259): #0      Object.noSuchMethod (dart:core/runtime/libobject_patch.dart:50:5)
I/flutter (10259): #1      _CupertinoPickerState._buildGradientScreen (package:flutter/src/cupertino/picker.dart:226:39)
I/flutter (10259): #2      _CupertinoPickerState.build (package:flutter/src/cupertino/picker.dart:331:9)
I/flutter (10259): #3      StatefulElement.build (package:flutter/src/widgets/framework.dart:3787:27)
I/flutter (10259): #4      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3699:15)
I/flutter (10259): #5      Element.rebuild (package:flutter/src/widgets/framework.dart:3547:5)
I/flutter (10259): #6      ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3679:5)
I/flutter (10259): #7      StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3826:11)
I/flutter (10259): #8      ComponentElement.mount (package:flutter/src/widgets/framework.dart:3674:5)
I/flutter (10259): #9      Element.inflateWidget (package:flutter/src/widgets/framework.dart:2950:14)
I/flutter (10259): #10     Element.updateChild (package:flutter/src/widgets/framework.dart:2753:12)
I/flutter (10259): #11     SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4838:14)
I/flutter (10259): #12     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2950:14)
I/flutter (10259): #13     Element.updateChild (package:flutter/src/widgets/framework.dart:2753:12)
I/flutter (10259): #14     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3710:16)
I/flutter (10259): #15     Element.rebuild (package:flutter/src/widgets/framework.dart:3547:5)
I/flutter (10259): #16     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3679:5)
I/flutter (10259): #17     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3674:5)
I/flutter (10259): #18     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2950:14)
I/flutter (10259): #19     MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4943:32)
I/flutter (10259): #20     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2950:14)
I/flutter (10259): #21     Element.updateChild (package:flutter/src/widgets/framework.dart:2753:12)
I/flutter (10259): #22     SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4838:14)
I/flutter (10259): #23     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2950:14)
I/flutter (10259): #24     Element.updateChild (package:flutter/src/widgets/framework.dart:2753:12)
I/flutter (10259): #25     SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4838:14)
I/flutter (10259): #26     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2950:14)
I/flutter (10259): #27     Element.updateChild (package:flutter/src/widgets/framework.dart:2753:12)
I/flutter (10259): #28     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3710:16)
I/flutter (10259): #29     Element.rebuild (package:flutter/src/widgets/framework.dart:3547:5)
I/flutter (10259): #30     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3679:5)
I/flutter (10259): #31     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3674:5)
I/flutter (10259): #32     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2950:14)
I/flutter (10259): #33     MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4943:32)
I/flutter (10259): #34     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2950:14)
I/flutter (10259): #35     Element.updateChild (package:flutter/src/widgets/framework.dart:2753:12)
I/flutter (10259): #36     SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4838:14)
I/flutter (10259): #37     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2950:14)
I/flutter (10259): #38     Element.updateChild (package:flutter/src/widgets/framework.dart:2753:12)
I/flutter (10259): #39     SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4838:14)
I/flutter (10259): #40     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2950:14)
I/flutter (10259): #41     Element.updateChild (package:flutter/src/widgets/framework.dart:2753:12)
I/flutter (10259): #42     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3710:16)
I/flutter (10259): #43     Element.rebuild (package:flutter/src/widgets/framework.dart:3547:5)
I/flutter (10259): #44     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3679:5)
I/flutter (10259): #45     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3674:5)
I/flutter (10259): #46     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2950:14)
I/flutter (10259): #47     Element.updateChild (package:flutter/src/widgets/framework.dart:2753:12)
I/flutter (10259): #48     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3710:16)
I/flutter (10259): #49     Element.rebuild (package:flutter/src/widgets/framework.dart:3547:5)
I/flutter (10259): #50     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3679:5)
I/flutter (10259): #51     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3674:5)
I/flutter (10259): #52     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2950:14)
I/flutter (10259): #53     Element.updateChild (package:flutter/src/widgets/framework.dart:2753:12)
I/flutter (10259): #54     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3710:16)
I/flutter (10259): #55     Element.rebuild (package:flutter/src/widgets/framework.dart:3547:5)
I/flutter (10259): #56     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3679:5)
I/flutter (10259): #57     StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3826:11)
I/flutter (10259): #58     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3674:5)
I/flutter (10259): #59     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2950:14)
I/flutter (10259): #60     Element.updateChild (package:flutter/src/widgets/framework.dart:2753:12)
I/flutter (10259): #61     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3710:16)
I/flutter (10259): #62     Element.rebuild (package:flutter/src/widgets/framework.dart:3547:5)
I/flutter (10259): #63     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3679:5)
I/flutter (10259): #64     StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3826:11)
I/flutter (10259): #65     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3674:5)
I/flutter (10259): #66     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2950:14)
I/flutter (10259): #67     Element.updateChild (package:flutter/src/widgets/framework.dart:2753:12)
I/flutter (10259): #68     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3710:16)
I/flutter (10259): #69     Element.rebuild (package:flutter/src/widgets/framework.dart:3547:5)
I/flutter (10259): #70     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3679:5)
I/flutter (10259): #71     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3674:5)
I/flutter (10259): #72     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2950:14)
I/flutter (10259): #73     Element.updateChild (package:flutter/src/widgets/framework.dart:2753:12)
I/flutter (10259): #74     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3710:16)
I/flutter (10259): #75     Element.rebuild (package:flutter/src/widgets/framework.dart:3547:5)
I/flutter (10259): #76     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3679:5)
I/flutter (10259): #77     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3674:5)
I/flutter (10259): #78     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2950:14)
I/flutter (10259): #79     MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4943:32)
I/flutter (10259): #80     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2950:14)
I/flutter (10259): #81     Element.updateChild (package:flutter/src/widgets/framework.dart:2753:12)
I/flutter (10259): #82     SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4838:14)
I/flutter (10259): #83     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2950:14)
I/flutter (10259): #84     Element.updateChild (package:flutter/src/widgets/framework.dart:2753:12)
I/flutter (10259): #85     SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4838:14)
I/flutter (10259): #86     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2950:14)
I/flutter (10259): #87     Element.updateChild (package:flutter/src/widgets/framework.dart:2753:12)
I/flutter (10259): #88     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3710:16)
I/flutter (10259): #89     Element.rebuild (package:flutter/src/widgets/framework.dart:3547:5)
I/flutter (10259): #90     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3679:5)
I/flutter (10259): #91     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3674:5)
I/flutter (10259): #92     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2950:14)
I/flutter (10259): #93     Element.updateChild (package:flutter/src/widgets/framework.dart:2753:12)
I/flutter (10259): #94     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3710:16)
I/flutter (10259): #95     Element.rebuild (package:flutter/src/widgets/framework.dart:3547:5)
I/flutter (10259): #96     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3679:5)
I/flutter (10259): #97     ComponentElement.mount (package:flutter/src/widgets/framework.dart:3674:5)
I/flutter (10259): #98     ParentDataElement.mount (package:flutter/src/widgets/framework.dart:4025:11)
I/flutter (10259): #99     Element.inflateWidget (package:flutter/src/widgets/framework.dart:2950:14)
I/flutter (10259): #100    Element.updateChild (package:flutter/src/widgets/framework.dart:2753:12)
I/flutter (10259): #101    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3710:16)
I/flutter (10259): #102    Element.rebuild (package:flutter/src/widgets/framework.dart:3547:5)
I/flutter (10259): #103    ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:3679:5)
I/flutter (10259): #104    StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3826:11)
I/flutter (10259): #105    ComponentElement.mount (package:flutter/src/widgets/framework.dart:3674:5)
I/flutter (10259): #106    Element.inflateWidget (package:flutter/src/widgets/framework.dart:2950:14)
I/flutter (10259): #107    Element.updateChild (package:flutter/src/widgets/framework.dart:2753:12)
I/flutter (10259): #108    SliverMultiBoxAdaptorElement.updateChild (package:flutter/src/widgets/sliver.dart:1012:36)
I/flutter (10259): #109    SliverMultiBoxAdaptorElement.createChild.<anonymous closure> (package:flutter/src/widgets/sliver.dart:997:20)
I/flutter (10259): #110    BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2266:19)
I/flutter (10259): #111    SliverMultiBoxAdaptorElement.createChild (package:flutter/src/widgets/sliver.dart:990:11)
I/flutter (10259): #112    RenderSliverMultiBoxAdaptor._createOrObtainChild.<anonymous closure> (package:flutter/src/rendering/sliver_multi_box_adaptor.dart:256:23)
I/flutter (10259): #113    RenderObject.invokeLayoutCallback.<anonymous closure> (package:flutter/src/rendering/object.dart:1730:58)
I/flutter (10259): #114    PipelineOwner._enableMutationsToDirtySubtrees (package:flutter/src/rendering/object.dart:799:15)
I/flutter (10259): #115    RenderObject.invokeLayoutCallback (package:flutter/src/rendering/object.dart:1730:13)
I/flutter (10259): #116    RenderSliverMultiBoxAdaptor._createOrObtainChild (package:flutter/src/rendering/sliver_multi_box_adaptor.dart:245:5)
I/flutter (10259): #117    RenderSliverMultiBoxAdaptor.addInitialChild (package:flutter/src/rendering/sliver_multi_box_adaptor.dart:323:5)
I/flutter (10259): #118    RenderSliverList.performLayout (package:flutter/src/rendering/sliver_list.dart:78:12)
I/flutter (10259): #119    RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (10259): #120    RenderSliverPadding.performLayout (package:flutter/src/rendering/sliver_padding.dart:182:11)
I/flutter (10259): #121    RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (10259): #122    RenderViewportBase.layoutChildSequence (package:flutter/src/rendering/viewport.dart:405:13)
I/flutter (10259): #123    RenderViewport._attemptLayout (package:flutter/src/rendering/viewport.dart:1316:12)
I/flutter (10259): #124    RenderViewport.performLayout (package:flutter/src/rendering/viewport.dart:1234:20)
I/flutter (10259): #125    RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (10259): #126    _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (10259): #127    RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (10259): #128    _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (10259): #129    RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (10259): #130    _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (10259): #131    RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (10259): #132    _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (10259): #133    RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (10259): #134    _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (10259): #135    RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (10259): #136    _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (10259): #137    RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (10259): #138    _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (10259): #139    RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (10259): #140    _RenderProxyBox&RenderBox&RenderObjectWithChildMixin&RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:104:13)
I/flutter (10259): #141    RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (10259): #142    RenderFlex.performLayout (package:flutter/src/rendering/flex.dart:798:17)
I/flutter (10259): #143    RenderObject.layout (package:flutter/src/rendering/object.dart:1634:7)
I/flutter (10259): #144    MultiChildLayoutDelegate.layoutChild (package:flutter/src/rendering/custom_layout.dart:142:11)
I/flutter (10259): #145    _ScaffoldLayout.performLayout (package:flutter/src/material/scaffold.dart:339:7)
I/flutter (10259): #146    MultiChildLayoutDelegate._callPerformLayout (package:flutter/src/rendering/custom_layout.dart:212:7)
I/flutter (10259): #147    RenderCustomMultiChildLayoutBox.performLayout (package:flutter/src/rendering/custom_layout.dart:356:14)
I/flutter (10259): #148    RenderObject._layoutWithoutResize (package:flutter/src/rendering/object.dart:1509:7)
I/flutter (10259): #149    PipelineOwner.flushLayout (package:flutter/src/rendering/object.dart:768:18)
I/flutter (10259): #150    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:281:19)
I/flutter (10259): #151    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding&WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:677:13)
I/flutter (10259): #152    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&SemanticsBinding&RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:219:5)
I/flutter (10259): #153    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:990:15)
I/flutter (10259): #154    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:930:9)
I/flutter (10259): #155    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:842:5)
I/flutter (10259): #156    _invoke (dart:ui/hooks.dart:151:13)
I/flutter (10259): #157    _drawFrame (dart:ui/hooks.dart:140:3)
I/flutter (10259): ════════════════════════════════════════════════════════════════════════════════════════════════════
```

```
$ flutter doctor -v
[√] Flutter (Channel beta, v0.11.3, on Microsoft Windows [Version 10.0.17134.407], locale de-DE)
    • Flutter version 0.11.3 at C:\SDKs\flutter
    • Framework revision 72bf075e8d (5 days ago), 2018-11-09 20:36:17 -0800
    • Engine revision 5646e86a6f
    • Dart version 2.1.0 (build 2.1.0-dev.9.3 9c07fb64c4)

[√] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at C:\SDKs\android
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = C:\SDKs\android
    • Java binary at: D:\Programme\JDK\bin\java
    • Java version Java(TM) SE Runtime Environment (build 1.8.0_92-b14)
    • All Android licenses accepted.

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.io/setup/#android-setup for detailed instructions).

[!] IntelliJ IDEA Ultimate Edition (version 2018.2)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA 2018.2.5
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.io/intellij-setup/#installing-the-plugins

[√] VS Code (version 1.29.0)
    • VS Code at C:\Users\puelo\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 2.20.0

[√] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 9 (API 28) (emulator)

! Doctor found issues in 2 categories.
```

**Comment by dnfield** [comment]
@jslavitz  or @xster might know more, but I suspect we need to default to the prior implementation in the case where background color is `null`.

**Comment by jslavitz** [comment]
I'll submit at PR on Monday, but for a quick fix, you want to change line #215 in picker.dart to 
`    if (widget.backgroundColor == null || widget.backgroundColor.alpha < 255)`. Sorry about the inconvenience!

**Comment by KammererTob** [comment]
Another quick fix is just to provide Colors.transparent as backgroundColor. No difference visually, but as noted a small performance impact.
